subprojects/packagefiles/nanobind/meson.build

```diff
@@ -52,6 +52,13 @@ if get_option('buildtype') == 'release'
   add_project_arguments('-DNB_COMPACT_ASSERTIONS', language: 'cpp')
 endif
 
+freethreaded = py.get_variable('Py_GIL_DISABLED', 0) != 0
+if freethreaded
+  freethreaded_def = '-DNB_FREE_THREADED'
+  add_project_arguments(freethreaded_def, language : 'cpp')
+  dep_compile_args += [freethreaded_def]
```

Comment:
Does this need to be passed through the dependency? I was under the impression it was only needed internally by nanobind

Reply:
This was a guess based on looking at the nanobind source. The main header, `nanobind/nanobind.h`, transitively includes headers like `nb_types.h` that make use of `NB_FREE_THREADED`, e.g. [here](https://github.com/wjakob/nanobind/blob/fd22b8c457cd291db2774a5526b5be39376212cd/include/nanobind/nb_types.h#L829)

Reply:
Ah interesting...I think that makes sense then